cups/ppd-private.h

```diff
@@ -214,6 +229,8 @@ extern const char	*_pwgMediaTypeForType(const char *media_type,
 extern const char	*_pwgPageSizeForMedia(pwg_media_t *media,
 			                      char *name, size_t namesize) _CUPS_PRIVATE;
 
+extern void             _ppdCacheAssignPresets(ppd_file_t *ppd, _ppd_cache_t *pc) _CUPS_PRIVATE;
```

Comment:
I don't think we want this here. cupsMarkOptions normally handles the PPD mapping, so let's do it there?

Reply:
cupsMarkOption(s) could do such an assignment, but it is much faster and more efficient if the presets are pre-generated in the PPD cache, as PPDs are static, they do not change from job to job. With the presets in the PPD cache, a `cupsMarkOptions()` on `print-color-mode=gray print-quality=5` would go very quickly.

Reply:
"gray" is not a valid IPP keyword for "print-color-mode"... The issue here is NOT speed (marking is not a significant cost, generating the PPD cache *is*), it is simplicity and reliability so that we can accurately map from IPP attribute names/values to PPD options/choices and back again.

Reply:
Sorry, it must be `monochrome`. Am I right?

Do you mean that if I create a `cups_option_t` list of `print-color-mode=monochrome print-quality=5` that `cupsMarkOptions()` called with this option list should mark the option settings according to the the preset for monochrome and high quality?